include/Inventory/Spell.h

```diff
@@ -0,0 +1,49 @@
+//
+// Created by musique88 on 8/22/20.
+//
+#pragma once
+#include <vector>
+#include "Inventory.h"
+
+namespace CC
+{
+
+	enum Element
+	{
+		NEUTRAL_ELEMENT,
+		WATER_ELEMENT,
+		FIRE_ELEMENT,
+		NATURE_ELEMENT,
+		ELECTRICITY_ELEMENT,
+		AIR_ELEMENT,
+		ICE_ELEMENT
+	};
+
+	enum Effect
+	{
+		STRONG_HEAL_EFFECT,
+		HEAL_EFFECT,
+		IMMUNE_EFFECT,
+		WEAK_EFFECT,
+		NORMAL_EFFECT,
+		STRONG_EFFECT
+	};
+
+	struct ElementRelationship
+	{
+		static Effect Relationship(Element attack, Element target);
+	};
+
+	class Spell
+	{
+	public:
+		Spell(Element element, std::array<int,NUMBER_OF_ELEMENTS> requirements);
+
+		Element element = NEUTRAL_ELEMENT;
```

Comment:
`element_`? To avoid the this->element = element you have in the Spell constructor.

Reply:
Requested changes has been done